pkg/merger/merger_test.go

```diff
@@ -408,6 +408,71 @@ func Test_MergeAndUpdate(t *testing.T) {
 		})
 	}
 }
+func Test_RecordLastModified(t *testing.T) {
+	cases := []struct {
+		name   string
+		attrs  *storage.ReaderObjectAttrs
+		mets   *Metrics
+		source string
+	}{
+		{
+			name:   "nil attrs; succeeds",
+			attrs:  nil,
+			mets:   &Metrics{},
+			source: "fake-source",
+		},
+		{
+			name:   "nil mets; succeeds",
+			attrs:  &storage.ReaderObjectAttrs{},
+			mets:   nil,
+			source: "fake-source",
+		},
+		{
+			name:   "nil; succeeds",
+			attrs:  nil,
+			mets:   nil,
+			source: "",
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			recordLastModified(tc.attrs, tc.mets, tc.source)
+		})
+	}
+}
+
+func Test_RecordLastModified_RecordMetrics(t *testing.T) {
+	var fakeMetric FakeInt64
+
+	attrs := &storage.ReaderObjectAttrs{
```

Comment:
feel free to ignore, but personally I prefer the following style:

```golang

attrs := storage.ReaderObjectAttrs{ 
  // etc
}

recordLastModified(&attrs, etc)
```

AKA declare it as a non-pointer when possible, and use its `&value` , rather than declaring it as a pointer.